bloomfilter/Data/BloomFilter/Mutable.hs

```diff
@@ -77,25 +75,18 @@ import Prelude hiding (elem, length, notElem,
 new :: Int                    -- ^ number of hash functions to use
     -> Int                    -- ^ number of bits in filter
     -> ST s (MBloom s a)
-new hash numBits = MB hash numBits' `liftM` V.new numBits'
-  where numBits' | numBits < 64 = 64
-                 | numBits > maxHash = maxHash
-                 | otherwise = ceil64 numBits
-              
-maxHash :: Int
-#if WORD_SIZE_IN_BITS == 64
-maxHash = 4294967296
-#else
-maxHash = 1073741824
-#endif
+new hash numBits = MB hash (fromIntegral numBits') `liftM` V.new numBits'
+  where numBits' | numBits < 64                = 64
+                 | numBits >= 0xffff_ffff_ffff = 0x1_0000_0000_0000
+                 | otherwise                   = ceil64 numBits
```

Comment:
In the PR you say

> We need to be more careful with remainder operations, we do them on unsigned Word64s.

> I limit the size of bloomfilter to 2^48; which is hopefully enough.

I think it's worth sticking that comment in the code here and explaining why. I presume you're talking about the problem that as one gets filters near to the hash size, the distribution of use of bits becomes increasingly uneven. And then the idea is that 16 bits should be enough to give good wraparound behaviour and a uniform distribution (with the difference in utilisation being on the order of 1/2^16).

Reply:
> presume you're talking about the problem that as one gets filters near to the hash size, the distribution of use of bits becomes increasingly uneven. 

Nope.

It's just an arbitrary bound. (On my machine `address sizes	: 43 bits physical, 48 bits virtual`, and 2^48 is a nice "round" number).